src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java

```diff
@@ -230,6 +218,47 @@ private String getConfigXml(Run build) {
     return null;
   }
 
+  private String updateScanStatus(String scanId, int status, int step, String results, TaskListener listener) throws IOException{
+    Retrofit client =
+            CwpClient.getClient(Configuration.get(), Jenkins.get().proxy, resolveCredentials());
+    CWPService cwpService = client.create(CWPService.class);
+    UpdateScanRequest request = new UpdateScanRequest();
+    try {
+      request.setStatus(status);
+      request.setStep(step);
+      JSONObject jsonObject = new JSONObject(results);
+      JSONObject resultsBlock = jsonObject.getJSONObject("results");
+      request.setViolationsCount(resultsBlock.optJSONArray("violations").length());
+      request.setPassedCount(resultsBlock.optJSONArray("passed_rules").length());
+      request.setSkippedCount(resultsBlock.optJSONArray("skipped_violations").length());
+    }catch(Exception e){
+      listener
+              .getLogger()
+              .println("Failed to form scan update request due to - " + e.getMessage());
+    }
+    Response<ScanResponse> response = cwpService.updateScanStatus(scanId,request).execute();
+    return getScanResponse(response);
+  }
+
+  private String getScanResponse(Response<ScanResponse> response) throws IOException {
+    if (response.code() == 200) {
```

Comment:
Best practice is to compare with http status code .

Reply:
Changes done and uploaded.